.github/workflows/doc.nuxeo.com.yml

```diff
@@ -0,0 +1,66 @@
+name: Build and Deploy doc.nuxeo.com
+
+on:
+  push:
+    branches:
+      - master
+      - development
+
+jobs:
+  set-environment:
+    runs-on: doc
+
+    environment:
+      name: ${{ github.ref == 'refs/heads/master' && 'production' || github.ref == 'refs/heads/development' && 'development' || '' }}
+  
+  build-and-deploy:
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '14'
+
+      - name: Check Versions and Save to File
+        run: |
+          echo "git $(git --version)" > runtime-versions.txt
+          echo "node version $(node --version)" >> runtime-versions.txt
+          echo "npm version $(npm --version)" >> runtime-versions.txt
+          echo "IP $(curl -s ipinfo.io/ip)" >> runtime-versions.txt
+
+      - name: Install dependencies
+        run: npm ci --registry="https://registry.npmjs.org" || npm install --registry="https://registry.npmjs.org"
+
+      - name: Build
+        run: ALGOLIA_KEY=${{ secrets.NX_TOKEN }} DEBUG_FILE_LEVEL='debug' npm start
+
+      - name: Generate Nginx Redirects
+        run: |
+            bin/redirects-yml2nginx > redirects.txt
+            echo > redirects_pageid.txt 
+
+      - name: Deploy to Server
+        with:
+          SERVER_SSH_PRIVATE_KEY: ${{ secrets.SERVER_SSH_PRIVATE_KEY }}
```

Comment:
How is that used? I see no option in th rsync command line to use that and not previous task seem to set the private key up for that job.